src/sqlparser/src/ast/mod.rs

```diff
@@ -3120,6 +3129,15 @@ impl fmt::Display for DiscardType {
     }
 }
 
+impl Statement {
+    pub fn to_redacted_string(&self) -> String {
+        REDACT_SQL_OPTIONS.set(true);
+        let redacted = self.to_string();
+        REDACT_SQL_OPTIONS.set(false);
```

Comment:
Can use `tokio::task_local` to automatically unset the flag when it goes out of the scope.

Reply:
> Can use tokio::task_local to automatically unset the flag when it goes out of the scope.

I used thread_local to avoid introducing a tokio dependency in risingwave_sqlparser crate.

I'm also fine with using `tokio::task_local`.

Reply:
fixed.